enhancements/ingress/set_eip_nlb_ingress.md

```diff
@@ -0,0 +1,936 @@
+    // https://docs.aws.amazon.com/AWSEC2/latest/UserGuide/elastic-ip-addresses-eip.html
+    // https://kubernetes-sigs.github.io/aws-load-balancer-controller/v2.7/guide/service/annotations/#eip-allocations
+    // +openshift:enable:FeatureGate=SetEIPForNLBIngressController
+    EIPAllocations []configv1.EIPAllocations `json:"eipAllocations"`
+}
+
+```
+
+### Workflow Description
+
+Pre-requisites: One must allocate an Elastic IP address from [Amazon's pool of public IPv4 addresses](https://docs.aws.amazon.com/AWSEC2/latest/UserGuide/elastic-ip-addresses-eip.html#using-instance-addressing-eips-allocating),
+or from a custom IP address pool that you have brought to your AWS account. For more information about bringing your own IP address range to your 
+AWS account, see [Bring your own IP addresses (BYOIP) in Amazon EC2](https://docs.aws.amazon.com/AWSEC2/latest/UserGuide/ec2-byoip.html).
+which could be then used in the `eipAllocations` field of the IngressController CR.
+Note: The AWS account has some cap on the number of EIPs per region. By default, all AWS accounts have a quota of five (5) Elastic IP addresses per Region, because public (IPv4) 
+internet addresses are a scarce public resource. For more details please check [Elastic IP address quota](https://docs.aws.amazon.com/AWSEC2/latest/UserGuide/elastic-ip-addresses-eip.html#using-instance-addressing-limit).
+When this cap is hit you will get the following error: `Elastic IP address could not be allocated.
+The maximum number of addresses has been reached.`
+Elastic IP Allocation can only be done for internet-facing NLB.
+
+**cluster administrator** is a human user responsible for deploying a
+cluster. A cluster administrator also has the rights to modify the cluster level components.
+
+#### Set EIP using an installer:
+1. Cluster administrator creates `install-config.yaml` and adds their domain, AWS region, and EIP allocation ids.
+```yaml
+apiVersion: v1
+baseDomain: example.com
+metadata:
+  name: test-cluster
+platform:
+  aws:
+    region: <AWS region>
+    lbType: NLB
+    networkLoadBalancer:
+      eipAllocations:
+      - eipalloc-1111
+      - eipalloc-2222
+      - eipalloc-3333
+      - eipalloc-4444
+      - eipalloc-5555 
+...
+```  
+Here, the administrator needs to know how many public subnets the `AWS VPC ID` which is tagged by `cluster name` after installation.
+To check VPC ID:
+Go to `Your VPCs` under `Virtual Public CLoud` section, then search for the VPC ID tagged with `cluster name` using the 
+command `oc get infrastructure cluster -o jsonpath='{.status.infrastructureName}'`
+To check the public subnets for the `VPC ID` tagged with the cluster name we got from the above command:
+Go the `AWS Cloud Console` click the `Load Balancers` tab then under the `details` click `VPC ID` under VPC, then click the link under `Main route table` and then check the column `Explicit subnet associations` which
+will show you the number of public subnets available for that VPC ID.
```

Comment:
I think this could benefit from some organization updates, and I am struggling to repeat this procedure.

I just tried it out, and I didn't have a "Load Balancers" tab, but I wasn't sure why you needed that? You found the VPC, just click the VPC ID link, and you at the "Main Route Table". I don't think you need anything about a load balancer, that seemed like an extra step that can just be removed:

```suggestion
To locate the public subnets:
1. Navigate to your AWS Web Console for the account that contains the cluster
2. Go to `Your VPCs` under the `Virtual Public Cloud` section
3. Search for the VPC ID tagged with `cluster name` using the command `oc get infrastructure cluster -o jsonpath='{.status.infrastructureName}'`
4. Click the VPC ID link for the VPC
5. Click the link under `Main route table` and then check the column `Explicit subnet associations` which
will show you the number of public subnets available for that VPC ID.
```

Reply:
ok done